microsetta_interface/implementation.py

```diff
@@ -1450,12 +1607,30 @@ def render_consent_page(account_id, source_id, form_type, sample_ids=None):
         participant_name=source_output['source_name'],
         age_range=source_output['consent']['age_range'],
         account_id=account_id,
-        source_id=source_id
+        source_id=source_id,
+        skip_dupe_check=skip_dupe_check
+    )
+
+
+def decline_reconsent(*, account_id, source_id):
+    # Add a variable to the session scope so the user can see their existing
+    # information but not add or update anything
+    session[RECONSENT_DECLINED_KEY] = True
+
+    return redirect(
+        "/accounts/%s/sources/%s" %
+        (account_id, source_id)
     )
 
 
 @prerequisite([SOURCE_PREREQS_MET, BIOSPECIMEN_PREREQS_MET])
 def get_source(*, account_id=None, source_id=None):
+    # If the user is switching profiles, we need to remove the
+    # reconsent_declined variable from the session scope if it exists
+    if session.get("source_id", False) and session["source_id"] != source_id\
```

Comment:
```suggestion
    if SOURCE_ID in session:
        if session[SOURCE_ID] != source_id:
            if session.get(RECONSENT_DECLINED_KEY, False):
```

minor but this avoids the line split. I'm not sure if the session key for source ID is stored already, if not would be nice for consistency

Reply:
Added SOURCE_ID as a constant and adjusted throughout implementation.py, along with the adjustment to avoid line splitting.